### ISSUE TITLE
When network changes, playback stops immediately

### ISSUE THREAD

**Issue by ghost** [lukedirtwalker/cutespotify#9]
When the network changes from WLAN to mobile network or vice versa, the music playback just stops immediately and it loses the playback point. This is painful, when I move away or into the range of any WLAN networks. This happens immediately every time when the connection changes.

The music must have been buffered enough so there shouldn’t be a problem. And even if the playback pauses, I think it should stay at the same song position and not just stop all playback and lose the current playlist.

There is no problem with offline tracks.

PS. The official Android version of Spotify does not have this problem on Jolla.

**Comment by lukedirtwalker** [comment]
Thanks I will look into this, currently I don't have any time.

**Comment by BonoNL** [comment]
I can confirm this, very annoying when driving in a car, it then happens often and you dont want to take the phone and restart the app while driving the highway.

**Comment by awlx** [comment]
Yep, can still confirm this. That's why I switched back to the Android app in the car :-/.

**Comment by lukedirtwalker** [comment]
@Sami-Perkele @BonoNL @awlx I investigated this issue and made a workaround: can you please test [this version](https://drive.google.com/file/d/0B-KOnAfO-Ux8Mk1iSjNOR1VULU0/view?usp=sharing) (under real world conditions) and tell me if it works?

**Comment by ghost** [comment]
Thanks for looking into this :) Seems to work great so far. But I tested very briefly. I will try to use it more tomorrow.

**Comment by awlx** [comment]
I will test it tomorrow when I drive a lot. Let's see if it works :).

**Comment by lukedirtwalker** [comment]
@Sami-Perkele @awlx @BonoNL  How did the tests go?

**Comment by ghost** [comment]
I haven't run in to any problems so far. I tried a couple of times. I started listening at home, with wi-fi, and then moving out of wifi range. Music continued playing as it should until the end of the playlist.

**Comment by awlx** [comment]
Can confirm it's fixed with the patch.

**Comment by BonoNL** [comment]
Luke, sorry for the not responding. I'm not daily on Github and the notifiction mails did't got it to my inbox. But I tested the latest version from Openrepos and it works a lot better!

I do noticed when driving like an hour in the Netherlands I always have hick ups on around the same area's. I've noticed then when the Jolla is losing the internet connection and is being reconnected Cutespot stays offline. And I need to restart the app. Maybe a larger buffer and reconnecting can fix this. I will try the version you give above Monday and Thursday.

**Comment by BonoNL** [comment]
Luke,

I've tested some weeks and have come to the following conclusion.
When the phone is losing internet connection for a short period, it's no problem. The buffer is long enough to hold so the phone can reconnect. But some times(mostly long car drives, 1h+) when the phone loses let's say connection multiple times along the ride the phone re-connects but CuteSpot isn't reconnecting after the third time or so. So I need to restart CuteSpot te acces my non offline cached music.

To be sure it wasn't a Sailfish/Jolla/network problem I tried the same route multiple times with CuteSpot and the Android Spotify version.
The Android version didn't have the problem.

The phone is connected through bluetooth with my Sony Xplod car stereo for playback.

**Comment by BonoNL** [comment]
Still having this issue.

**Comment by lukedirtwalker** [comment]
Reopened since CuteSpot goes into offline mode when it shouldn't
